template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/oozie/OozieRoleConfigProvider.java

```diff
@@ -0,0 +1,111 @@
+package com.sequenceiq.cloudbreak.cmtemplate.configproviders.oozie;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.stereotype.Component;
+
+import com.cloudera.api.swagger.model.ApiClusterTemplateConfig;
+import com.cloudera.api.swagger.model.ApiClusterTemplateVariable;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.database.base.DatabaseType;
+import com.sequenceiq.cloudbreak.cmtemplate.CmTemplateProcessor;
+import com.sequenceiq.cloudbreak.cmtemplate.configproviders.AbstractRoleConfigConfigProvider;
+import com.sequenceiq.cloudbreak.domain.RDSConfig;
+import com.sequenceiq.cloudbreak.template.TemplatePreparationObject;
+import com.sequenceiq.cloudbreak.template.views.HostgroupView;
+import com.sequenceiq.cloudbreak.template.views.RdsView;
+
+@Component
+public class OozieRoleConfigProvider extends AbstractRoleConfigConfigProvider {
+
+    private static final String OOZIE_DATABASE_HOST = "oozie_database_host";
+
+    private static final String OOZIE_DATABASE_NAME = "oozie_database_name";
+
+    private static final String OOZIE_DATABASE_TYPE = "oozie_database_type";
+
+    private static final String OOZIE_DATABASE_USER = "oozie_database_user";
+
+    private static final String OOZIE_DATABASE_PASSWORD = "oozie_database_password";
+
+    @Override
+    protected List<ApiClusterTemplateConfig> getRoleConfig(String roleType, HostgroupView hostGroupView) {
+        List<ApiClusterTemplateConfig> roleConfigs = new ArrayList<>();
+
+        switch (roleType) {
+            case "OOZIE_SERVER":
+                String databaseHost = getRoleTypeVariableName(hostGroupView.getName(), roleType, OOZIE_DATABASE_HOST);
+                roleConfigs.add(new ApiClusterTemplateConfig().name(OOZIE_DATABASE_HOST).variable(databaseHost));
+
+                String databaseName = getRoleTypeVariableName(hostGroupView.getName(), roleType, OOZIE_DATABASE_NAME);
```

Comment:
Using the hostGroupView based template variable naming since in HA Mode, oozie_server can be deployed to multiple hosts.